### PR TITLE
Replaced basestring by six.string_types

### PR DESCRIPTION
**Pull request by claudep**
Also converted (str, unicode) to six.string_types (and some str which were supposed to be basestring).

### CODE REVIEW

File: translate/storage/placeables/strelem.py [hunk 1]
```diff
@@ -881,12 +882,12 @@ def prune(self):
                 # Remove empty strings or StringElem nodes
                 # (but not StringElem sub-class instances, because they
                 # might contain important (non-rendered) data.
-                if (type(elem.sub[i]) in (StringElem, str, unicode) and
-                    len(elem.sub[i]) == 0):
+                if ((type(elem.sub[i]) == StringElem or isinstance(elem.sub[i], six.string_types)) and
+                        len(elem.sub[i]) == 0):
```
Comment (unho) [inline translate/storage/placeables/strelem.py]: Please remove the added indent in this last line so it aligns with the parenthesis.

**Comment by unho** [comment]
Will merge once the concern is addressed and the PR is rebased.

File: translate/storage/placeables/strelem.py (hunk 1 above)
Comment (claudep) [inline translate/storage/placeables/strelem.py]: This is one of PEP8 proposal to indent conditionals (to not be visually aligned with the following code). But I'm fine with keeping a single indentation, too.

**Comment by claudep** [comment]
Fixed and rebased.

**Comment by unho** [comment]
Please rebase again.

**Comment by claudep** [comment]
Done

**Comment by unho** [comment]
Merged. Thanks a lot!

**Pull request merged**